### DISCUSSION TITLE
Upgrading from 1.1.8 to 1.2.5

### DISCUSSION THREAD

**Discussion by tilenk-lucis** [wintercms/winter#1085] [Q&A]
Hello, I recently upgraded from WinterCMS version 1.1.8 to 1.2.5. Now I have a problem with one of my custom plugins, which I cannot seem to solve.

I have a vendor folder called `sp` which contains 3 plugins essentials, widgets and news. The widgets plugin tries to include the trait from the essentials plugin, but cannot seem to find the plugin folder, even though the entire  plugin was not changed during the upgrade. 

This is the Plugin.php in the widgets plugin.

    <?php namespace Sp\Widgets;

    use Cms\Classes\Controller;
    ...
    use Sp\Essentials\Traits\ComponentHelpers;

    class Plugin extends PluginBase
    {
        use ComponentHelpers;

        public $require = ['Sp.Essentials'];
        ...
    }

And this is my Trait file in Sp\Essentials\Traits location.

    <?php namespace Sp\Essentials\Traits;

    use ReflectionClass;
    use Cms\Classes\ComponentBase;

    trait ComponentHelpers
    {
        ....
    }

I tried manually including the plugin folder, but that just makes the aplication crash further down the line, when it tries doing something else with the plugin. The folder names are in lowercase, but from what I read, that should not be a problem. They worked fine before the upgrade too.

The problem it throws is this:

    Trait "Sp\Essentials\Traits\ComponentHelpers" not found

Edit:
Throught trial and error I found out, if I disable the call for the reflection code, the site works (some of the content still doesnt, since it is linked to the plugins).

    $reflection = new ReflectionProperty(ControllerBehaviorLibrary::class, 'instance');
    $reflection->setAccessible(true);
    $reflection->setValue(null, ControllerBehavior::instance());

Is it possible the reflection class changed from version 1.1?

**Comment by damsfx** [comment]
@mts-coder Have you tried switching the case of the trait's path?
I think the path need to be lowercase `./plugins/sp/essentials/traits`.

You can also try to use the trait without a `use` directive.
```php
<?php namespace Sp\Widgets;

use Cms\Classes\Controller;
...

class Plugin extends PluginBase
{
    use \Sp\Essentials\Traits\ComponentHelpers;

    public $require = ['Sp.Essentials'];
    ...
}
```

**Reply by tilenk-lucis** [reply]
I have the path in lowercase, but thought that since namespace is not all in lowercase, it might affect it. Switching between them still causes the same issue. 

I also tried including it like you provided, but it is also still the same.

**Reply by LukeTowers** [reply]
@mts-coder what is the output of `artisan plugin:list`? It's possible there's something preventing your essentials plugin from loading which would prevent the autoloader from knowing where to look for its classes.
